### PR TITLE
py-pip-tools: bump to latest 6.12.3 / fix steps deps

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

**Comment by tristan0x** [comment]
The audit found issues in other packages than py-pip-tools. Can I disable this precheck @matz-e  ?

**Comment by matz-e** [comment]
> The audit found issues in other packages than py-pip-tools. Can I disable this precheck @matz-e ?

it's not required. is that an issue?

**Comment by tristan0x** [comment]
> > The audit found issues in other packages than py-pip-tools. Can I disable this precheck @matz-e ?
> 
> it's not required. is that an issue?

not at all. But it looks like it skips the other pre-checks because this one failed.

File: bluebrain/repo-patches/packages/steps/package.py [hunk 1]
```diff
@@ -75,9 +75,9 @@ class Steps(CMakePackage):
     depends_on("py-gcovr", when="+coverage", type="build")
     depends_on("py-matplotlib", type=("build", "test"))
     depends_on("py-mpi4py", when="+distmesh", type=("build", "test", "run"))
-    depends_on("py-nose", when="@3:", type=("build", "test"))
+    depends_on("py-nose", when="@3:4", type=("build", "test"))
```
Comment (tristan0x) [inline bluebrain/repo-patches/packages/steps/package.py:78]: I found the dependency which forced the use of an old py-virtualenv! We don't need that anymore!

**Review by matz-e — APPROVED** [review]

**Comment by tristan0x** [comment]
Hey @matz-e 
The GitLab CI/CD job seems to not be triggered.

When started manually from GitLab WebUI, I have the error message : 

```
Pipeline cannot be run.
Pipeline filtered out by workflow rules.
```

What are these rules?

**Comment by olupton** [comment]
This branch is breaking the GitHub -> GitLab mirroring, for some reason...do you have any idea why @tristan0x ? (I assume you have access to look at the errors in GitLab)

**Comment by olupton** [comment]
> This branch is breaking the GitHub -> GitLab mirroring, for some reason...do you have any idea why @tristan0x ? (I assume you have access to look at the errors in GitLab)

Answering my own question, I think this is because of your old `py-pip-tools` branch @tristan0x. You could probably fix it by deleting that branch on GitLab.

**Comment by tristan0x** [comment]
> This branch is breaking the GitHub -> GitLab mirroring, for some reason...do you have any idea why @tristan0x ? (I assume you have access to look at the errors in GitLab)

I can see that in GitLab
```
Failed to create branch 'py-pip-tools/6.12.3': invalid reference name '7e3a92a466be4a36340305f74ec91da586f03702'```
```

I have just removed it from GitHub

**Comment by tristan0x** [comment]
The refresh has just succeeded @olupton

**Pull request closed without merging**